src/ethereum/cancun/vm/__init__.py

```diff
@@ -28,27 +28,40 @@
 
 
 @dataclass
-class Environment:
+class BlockEnvironment:
     """
     Items external to the virtual machine itself, provided by the environment.
     """
 
-    caller: Address
+    chain_id: U64
```

Comment:
I almost wish there was a `ChainEnvironment` or a reference to the genesis here instead of `chain_id`.

Reply:
Do you see any further chain specific attributes that could be added to this in the future? If not, just maintaining the `chain_id` as a special case attribute in `BlockEnvironment` for now makes sense to me.

Reply:
I guess the alternative would be including a `GenesisConfiguration`, which seems overkill for just the chain id.